articles/embedded.md

```diff
@@ -21,26 +21,53 @@ While standard ROS 2 can run on Linux-class embedded devices, smaller microcontr
+|-----|-------------|----------|
+| Q-RTOS | W-RTOS, W-PORTABILITY | Which RTOS(s) do we use as the basis? |
+| Q-BUILD | W-RTOS, W-PORTABILITY | How do we handle the RTOS(s) respective build-systems? |
+| Q-LANG | W-PORTABILITY | Which language should be used, and at what spec level? |
+| Q-API | W-PORTABILITY, W-MODULARITY | How should the API look in general?|
+| Q-PERF | W-PORTABILITY, W-DEVICES, W-CONTROL | What are the performance implications of the API?|
+| Q-MIDDLEWARE | W-MW | Which communication/middleware technology is used?|
+| Q-MCU-SPEC | W-DEVICES, W-MCU-SPEC, W-CONTROL | Which MCU-specific features do we offer in the external API?|
 
-*   **High Modularity:** The embedded ROS 2 stack shall be highly modular, not only to allow for adaptation to specific hardware features and different RTOS, but also to allow the integration with existing frameworks and to be able to create customized sub-stacks. For example, it shall be usable in the style of rosserial, thus bringing basically the message serialization to the MCU only. Similarly, it should be possible to derive a sub-stack that provides node graph features and publish/subscribe but no parameters and services, and so on.
+## Actions
 
-*   **MCU-specific core functionalities and mechanisms:** Finally, the embedded ROS 2 stack shall include novel core functionalities and mechanisms that are specific for MCUs. These include power management functionalities and mechanisms for static initialization and memory management.
+> *TODO: Maybe don't call this action, but "fact finding" or the like?*
 
-In the EU project OFERA, we compiled a longer list of requirements to an embedded ROS 2 stack, which might serve as a good basis for the discussion. Please see Section 5 in [ofera.eu/deliverables/D1.7_Requirements.pdf](http://ofera.eu/deliverables/D1.7_Requirements.pdf).
+To answer these questions, both the OFERA EU project as well as several others have already undertaken or are planning exploratory work.
+
+*Meta-Note*: Please only add a short description here, linking to more detailed pages if necessary.
 
+| Key | Related Question | Action | Description | Links |
+|-----|------------------|--------|-------------|-------|
+| A-RTOS | Q-RTOS | RTOS Proof-of-Concept | Provide a proof-of-concept RTOS. In the OFERA project, we chose NuttX, because it is largely POSIX compatible and thus eases porting. There are also experiments based on RIOT and FreeRTOS | TODO |
+| A-BUILD-META | Q-BUILD | Meta-Build | Explores a meta-build approach to transform ROS 2 CMakeLists.txt to RTOS-specific build instructions. | TODO |
+| A-BUILD-NUTTX | Q-BUILD | NuttX-specific build | OFERA has integrated (parts of) micro-ROS directly as an app in the NuttX build. |TODO |
+| A-BUILD-ANDROID | Q-BUILD | Arduino Build | Robotis has explored building all the libraries using the Arduino IDE. This required some manual changes and thus does not scale, but can get you off the ground. | TODO |
```

Comment:
`A-BUILD-ANDROID` should not it be `A-BUILD-ARDUINO`?

Reply:
Yep, done.